### PR TITLE
Updated mathjs

### PR DESCRIPTION
**Pull request by CaselIT**
case 2. Improvement to existing type definition.
http://mathjs.org/docs/expressions/expression_trees.html#properties
http://mathjs.org/docs/expressions/expression_trees.html#operatornode

Added additional properties to interface `MathNode` for ConstantNode and OperatorNode

### CODE REVIEW

File: mathjs/mathjs.d.ts [hunk 1]
```diff
@@ -1334,6 +1334,10 @@ declare namespace mathjs {
 	export interface MathNode {
                 isNode: boolean;
                 isSymbolNode: boolean;
+                isConstantNode: boolean;
```
Comment (mhegazy) [inline mathjs/mathjs.d.ts]: i do not see any reference to `isConstantNod` or `isOperatorNode` in the documentation. can you provide more details?

File: mathjs/mathjs.d.ts [hunk 2]
```diff
@@ -1334,6 +1334,10 @@ declare namespace mathjs {
...
+                isOperatorNode: boolean;
+                op: string;
```
Comment (mhegazy) [inline mathjs/mathjs.d.ts]: these are only visible on ConstantNode, and not all nodes. so either create a subclass, or make them optional.

Comment (CaselIT) [inline mathjs/mathjs.d.ts]: I followed the previous convention, also `isSymbolNode` `name` and `value` are only visible on some some node types.
But I agree that it's better to set them optional for clarity.

**Comment by CaselIT** [comment]
Thanks  @mhegazy

**Pull request merged**